addons/misra.py

```diff
@@ -1494,6 +1494,249 @@ def misra_8_14(self, rawTokens):
             if token.str == 'restrict':
                 self.reportError(token, 8, 14)
 
+    def misra_9_2(self, data):
+        # Holds information about a struct or union's element definition.
+        class ElementDef:
+            def __init__(self, elementType, name, valueType = None, dimensions = None):
+                self.elementType = elementType
+                self.name = name
+                self.valueType = valueType
+                self.dimensions = dimensions
+
+        def getIntvalue(token):
+            if token and token.values and len(token.values) > 0:
+                return token.values[0].intvalue
+            else:
+                return None
+
+        # Return an array containing the size of each dimension of an array declaration,
+        # or coordinates of a designator in an array initializer,
+        # and the name token's valueType, if it exist.
+        #
+        # In the examples below, the ^ indicates the initial token passed to the function.
+        #
+        # Ex:   int arr[1][2][3] = .....
+        #                    ^
+        #       returns: [1,2,3], valueType
+        #
+        # Ex:   int arr[3][4] = { [1][2] = 5 }
+        #                            ^
+        #       returns [1,2], None
+        def getArrayDimensionsAndValueType(token):
+            dimensions = []
+            while token and token.str == '[':
+                op1Value = getIntvalue(token.astOperand1)
```

Comment:
hmm.. do we need to handle variable length arrays?

    void foo(int x) {
        if (x == 10 || x == 100) {
             int arr[x];   // <- x has two values 10 and 100.
        }
    }

Reply:
For 9.2 we don't really use the length, since 9.2 only cares about structure. 9.3 and 9.4, however, will have to.

Reply:
I wonder what the intention is with `op1Value = getIntvalue(token.astOperand1)`...  do you want to handle code such as `10[arr]` that is legal but very unusual? Or is there some use case I don't see where operand1 is a int value?